mempool/src/shared_mempool/coordinator.rs

```diff
@@ -143,7 +186,12 @@ pub(crate) async fn request_coordinator<V>(
                                             .with_label_values(&["received".to_string().deref(), peer_id.to_string().deref()])
                                             .inc_by(transactions.len() as i64);
                                         let smp_clone = smp.clone();
-                                        let timeline_state = match peer_manager.is_upstream_peer(network_id, peer_id) {
+                                        let peer = PeerNetworkId(network_id, peer_id);
+                                        let timeline_state = match peer_manager
+                                            .lock()
```

Comment:
nit: would love to get rid of mutex around peer_manager,  but can we at least move it back to one level down as in previous version
In such case at least `is_upstream_peer` calls won't require a lock acquisition